### PR TITLE
Update icon and improve deb package

### PR DESCRIPTION
**Pull request by qparis**

### CODE REVIEW

**Comment by madoar** [comment]
Can you give us some more details how you improved the `deb` package?

**Comment by qparis** [comment]
The new deb packages is pretty much the same than POLv4 one:
* phoenicis goes inside /usr/share/phoenicis instead of /opt 
* 3 icons are added: (Full size, 32, 16). Are you ok with this one (until we have a new logo) 
phoenicis
* Shortcut is created 
* Launcher is placed inside /usr/bin

**Comment by madoar** [comment]
The logo looks good to me.
Do you have a `deb` for me to test?

File: phoenicis-dist/src/resources/Phoenicis.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.0
+Encoding=UTF-8
+Name=Phoenicis
+GenericName=Game
+Comment=Phoenicis
+Type=Application
+Exec=phoenicis %F
+Icon=phoenicis
+Categories=Game;
```
Comment (madoar) [inline phoenicis-dist/src/resources/Phoenicis.desktop]: I don't like the category.
I mean `Phoenicis` is not really a game or a game management application it is much more generic.

File: phoenicis-dist/src/scripts/phoenicis-create-package.sh [hunk 2]
```diff
@@ -62,7 +62,45 @@ fi
...
+
+    packageName="Phoenicis_$VERSION"
+    cd "$PHOENICIS_TARGET"
+    rmdir packages/PhoenicisPlayOnLinux/
+    rm -rf "packages/phoenicis" 2> /dev/null
+    mv packages/Phoenicis\ PlayOnLinux/ packages/phoenicis
+    rm -rf "$packageName" 2> /dev/null
+    mkdir -p "$packageName/DEBIAN/"
+
+    cat << EOF > "$packageName/DEBIAN/control"
```
Comment (madoar) [inline phoenicis-dist/src/scripts/phoenicis-create-package.sh]: Should we maybe move this to its own file?

**Comment by qparis** [comment]
www.playonlinux.com/Phoenicis_5.0-SNAPSHOT.deb

File: phoenicis-dist/src/scripts/phoenicis-create-package.sh [hunk 3]
```diff
@@ -62,7 +62,45 @@ fi
 
 if [ "$PHOENICIS_OPERATING_SYSTEM" == "Linux" ]; then
     jpackager create-image "${PHOENICIS_JPACKAGER_ARGUMENTS[@]}"  --linux-bundle-name "phoenicis-playonlinux"
-    jpackager create-installer deb "${PHOENICIS_JPACKAGER_ARGUMENTS[@]}" --linux-package-deps "unzip, wget, xterm | x-terminal-emulator, imagemagick, cabextract, icoutils, p7zip-full, curl, winbind" --linux-deb-maintainer "PlayOnLinux Packaging <packages@playonlinux.com>" --linux-bundle-name "phoenicis-playonlinux"
```
Comment (madoar) [inline phoenicis-dist/src/scripts/phoenicis-create-package.sh]: What about `rpm` installer?
As far as I understand `jpackager` it is also able to create `rpm` installer

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (qparis) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: Yep, any other suggestion?

**Comment by madoar** [comment]
I'll test the `deb` tomorrow.

File: phoenicis-dist/src/scripts/phoenicis-create-package.sh (hunk 3 above)
Comment (qparis) [reply phoenicis-dist/src/scripts/phoenicis-create-package.sh]: jpackager does no longer create the .deb. I do it manually, it's much more flexible.
The rpm will be done the same way as we did in for POLv4, maybe in a separate ticket

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (madoar) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: I think @plata had some additional categories in our old `.desktop` file.
I don't remember in detail which they were

File: phoenicis-dist/src/scripts/phoenicis-create-package.sh (hunk 2 above)
Comment (qparis) [reply phoenicis-dist/src/scripts/phoenicis-create-package.sh]: Not easily, the version is dynamic

Comment (qparis) [reply phoenicis-dist/src/scripts/phoenicis-create-package.sh]: (And also later the installed size)

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (ImperatorS79) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: You can put it in multiple category. Utility and games, for examples.

**Comment by ImperatorS79** [comment]
Seems to work for me. Here is lintian output:
```
E: phoenicis-playonlinux: arch-independent-package-contains-binary-or-object usr/share/phoenicis/PhoenicisPlayOnLinux
E: phoenicis-playonlinux: arch-dependent-file-in-usr-share usr/share/phoenicis/PhoenicisPlayOnLinux
E: phoenicis-playonlinux: unstripped-binary-or-object usr/share/phoenicis/PhoenicisPlayOnLinux
E: phoenicis-playonlinux: arch-independent-package-contains-binary-or-object usr/share/phoenicis/libpackager.so
E: phoenicis-playonlinux: arch-dependent-file-in-usr-share usr/share/phoenicis/libpackager.so
E: phoenicis-playonlinux: unstripped-binary-or-object usr/share/phoenicis/libpackager.so
E: phoenicis-playonlinux: arch-independent-package-contains-binary-or-object usr/share/phoenicis/runtime/bin/java
E: phoenicis-playonlinux: arch-dependent-file-in-usr-share usr/share/phoenicis/runtime/bin/java
E: phoenicis-playonlinux: unstripped-binary-or-object usr/share/phoenicis/runtime/bin/java
E: phoenicis-playonlinux: arch-independent-package-contains-binary-or-object ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: phoenicis-playonlinux: arch-dependent-file-in-usr-share ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: phoenicis-playonlinux: unstripped-binary-or-object ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: phoenicis-playonlinux: embedded-library usr/share/phoenicis/runtime/lib/libjavajpeg.so: libjpeg
E: phoenicis-playonlinux: embedded-library usr/share/phoenicis/runtime/lib/liblcms.so: lcms2
E: phoenicis-playonlinux: embedded-library usr/share/phoenicis/runtime/lib/libsplashscreen.so: libjpeg
E: phoenicis-playonlinux: embedded-library ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: phoenicis-playonlinux: missing-dependency-on-libc needed by usr/share/phoenicis/PhoenicisPlayOnLinux and 25 others
E: phoenicis-playonlinux: debian-changelog-file-missing
E: phoenicis-playonlinux: no-copyright-file
W: phoenicis-playonlinux: spelling-error-in-description softwares software
W: phoenicis-playonlinux: spelling-error-in-description differents different
W: phoenicis-playonlinux: non-standard-file-perm usr/share/phoenicis/runtime/legal/java.base/ADDITIONAL_LICENSE_INFO 0444 != 0644
W: phoenicis-playonlinux: non-standard-file-perm usr/share/phoenicis/runtime/legal/java.base/ASSEMBLY_EXCEPTION 0444 != 0644
W: phoenicis-playonlinux: non-standard-file-perm usr/share/phoenicis/runtime/legal/java.base/LICENSE 0444 != 0644
W: phoenicis-playonlinux: non-standard-file-perm ... use --no-tag-display-limit to see all (or pipe to a file/program)
W: phoenicis-playonlinux: codeless-jar usr/share/phoenicis/app/javafx-base-11.0.1.jar
W: phoenicis-playonlinux: codeless-jar usr/share/phoenicis/app/javafx-controls-11.0.1.jar
W: phoenicis-playonlinux: codeless-jar usr/share/phoenicis/app/javafx-graphics-11.0.1.jar
W: phoenicis-playonlinux: codeless-jar ... use --no-tag-display-limit to see all (or pipe to a file/program)
W: phoenicis-playonlinux: binary-without-manpage usr/bin/phoenicis
```

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (plata) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: Yes, the previous control file had "Game;Utility" which is what I used now for the flatpak as well (see https://github.com/PhoenicisOrg/phoenicis/blob/master/phoenicis-dist/src/flatpak/org.phoenicis.playonlinux.appdata.xml).
Just realized that there is a mistake... The appdata.xml and the .desktop are using different categories. If we decide on the categories here, I can update that as well.

File: i18n/keys.pot [hunk 4]
```diff
@@ -17,32 +17,18 @@ msgstr ""
...
-
-#: phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/containers/ContainersController.java:71
-#: phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/containers/ContainersController.java:123
-msgid "Loading engine tools failed."
-msgstr ""
-
-#: phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/containers/ContainersController.java:90
-#, java-format
-msgid "Delete {0} container"
+#: phoenicis-tools/src/main/java/org/phoenicis/tools/checksum/ChecksumCalculator.java:39
```
Comment (plata) [inline i18n/keys.pot]: This file should not be part of the PR.

File: phoenicis-dist/src/launchers/phoenicis [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+/usr/share/phoenicis/PhoenicisPlayOnLinux "$@"
```
Comment (plata) [inline phoenicis-dist/src/launchers/phoenicis]: Add newline at EOF.

File: phoenicis-dist/src/resources/Phoenicis.desktop [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.0
+Encoding=UTF-8
+Name=Phoenicis
```
Comment (plata) [inline phoenicis-dist/src/resources/Phoenicis.desktop]: Shouldn't this be "Phoenicis PlayOnLinux" (after the discussion we had yesterday)?

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (plata) [inline phoenicis-dist/src/resources/Phoenicis.desktop]: Add newline at EOF.

File: i18n/keys.pot (hunk 4 above)
Comment (qparis) [reply i18n/keys.pot]: Maybe we should prevent maven from updating automatically

**Comment by qparis** [comment]
Regarding the icon, it was to have a better consistency between the website, PlayOnMac, and the rest.
Do you see any other places the old one is?

**Comment by qparis** [comment]
We will fix lintian problems in another ticket

**Comment by qparis** [comment]
Ready for review

**Comment by plata** [comment]
My comments regarding newline at EOF are still open. What about the categories for the .desktop? I think the PR should be renamed to "Update icon and improve deb package".

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (qparis) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: I think it is now updated isn't it?

File: phoenicis-dist/src/launchers/phoenicis (hunk 5 above)
Comment (qparis) [reply phoenicis-dist/src/launchers/phoenicis]: I think it is now updated isn't it?

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (plata) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: No. Check the diff. It shows this red symbol.

File: phoenicis-dist/src/launchers/phoenicis (hunk 5 above)
Comment (plata) [reply phoenicis-dist/src/launchers/phoenicis]: No. Check the diff. It shows this red symbol.

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (qparis) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: Does not seem so: https://github.com/PhoenicisOrg/phoenicis/pull/1670/files

**Comment by madoar** [comment]
I have the engines tab open for 15 - 20 minutes now and it still shows the progress indicator.
I don't think it is merely slow^^

**Comment by qparis** [comment]
> I have the engines tab open for 15 - 20 minutes now and it still shows the progress indicator.
> I don't think it is merely slow^^

I'm wondering if it's not a multithreading issue. As we have set only one thread for appsExecutorService and containersExecutorService, if for any reason once of this thread is blocked, it will lock the pool. Do you know in which pool does the wine webservice belongs to?

File: phoenicis-dist/src/launchers/phoenicis (hunk 5 above)
Comment (plata) [reply phoenicis-dist/src/launchers/phoenicis]: Ok. Fixed.

File: phoenicis-dist/src/resources/Phoenicis.desktop (hunk 1 above)
Comment (plata) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: Ok. Fixed.

**Comment by madoar** [comment]
It seems to use the script executor service:
```java
    @Bean
    public ControlledThreadPoolExecutorService scriptExecutorService() {
        return new ControlledThreadPoolExecutorService("Scripts", 10, 50);
    }
```

**Comment by plata** [comment]
@madoar but only with the deb?

**Comment by madoar** [comment]
I've tested only the `deb`, I can't build the current version myself currently because of my Java 8 issue.

**Comment by qparis** [comment]
I think it is more or less random. I also had this problem without the deb

**Comment by plata** [comment]
Then we should not care about this here.

**Comment by qparis** [comment]
It would be nice if someone could confirm that the deb is working. Also, note that it is a SNAPSHOT deb and not alpha-2 one

**Comment by plata** [comment]
@ImperatorS79 did that already.

**Comment by madoar** [comment]
How do you define working?
I mean I can install the `deb` (http://www.playonlinux.com/Phoenicis_5.0-SNAPSHOT.deb) and I can start it afterwards.
The issues I've found I mentioned above.

File: phoenicis-dist/src/resources/Phoenicis.desktop [hunk 7]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.0
+Encoding=UTF-8
+Name=Phoenicis PlayOnLinux
+GenericName=Game;Utility
```
Comment (plata) [inline phoenicis-dist/src/resources/Phoenicis.desktop]: This should be in `Categories` and not here.

Comment (qparis) [reply phoenicis-dist/src/resources/Phoenicis.desktop]: Done

**Comment by qparis** [comment]
> How do you define working?
> I mean I can install the `deb` (http://www.playonlinux.com/Phoenicis_5.0-SNAPSHOT.deb) and I can start it afterwards.
> The issues I've found I mentioned above.

I don't know if it is a packaging issue or a Phoenicis issue. In the later case, should it need to be fixed for alpha-2?

File: phoenicis-dist/src/flatpak/org.phoenicis.playonlinux.desktop [hunk 8]
```diff
@@ -7,6 +7,6 @@ Terminal=false
 Type=Application
 MimeType=application/x-ms-dos-executable;application/x-msi;application/x-ms-shortcut;
 Icon=org.phoenicis.playonlinux
-Categories=Utility
+Categories=Games;Utility
```
Comment (plata) [inline phoenicis-dist/src/flatpak/org.phoenicis.playonlinux.desktop]: Should be "Game" (see https://standards.freedesktop.org/menu-spec/latest/apa.html).

**Comment by plata** [comment]
I've never seen this issue in the engines tab.

**Comment by qparis** [comment]
I can, but I'd also like to add few memory optimization

**Comment by qparis** [comment]
@madoar If the problem comes from the .deb file, it does not come from this ticket anyway. I suggest merging it and I'll investigate on your issue
